app/src/main/java/net/squanchy/service/firebase/FirebaseDbService.java

```diff
@@ -143,4 +153,17 @@ private Completable updateFavorite(String eventId, Func1<DatabaseReference, Task
                     .addOnFailureListener(emitter::onError);
         });
     }
+
+    public Completable addAchievement(String userId, String achievementId, Long timestamp) {
+        return updateAchievement(userId, achievementId, reference -> reference.setValue(timestamp));
+    }
+
+    public Completable updateAchievement(String userId, String achievementId, Func1<DatabaseReference, Task<Void>> action) {
```

Comment:
Yes, since the user doesn't get to take back an achievement, I will simplify it down to just a method